tutorials/erb_template.md

```diff
@@ -0,0 +1,129 @@
+\page erbtemplate ERB Template
+
+This tutorial shows how to place a large number of entites in a simulation world using ERB. ERB is the commonly used templating language to generate structured text files with Ruby code. In this case, since most simulation world used in gazebo is defined with sdformat (XML documents), ERB works conveniently.
+
+## Set up Ruby
+
+Firstly, Ruby needs to be installed. If you have gone through [ignition libraries installation guide](https://ignitionrobotics.org/docs/latest/install), it's most likely you already have ruby installed. To check if Ruby is installed, use 
+```{.sh}
+ruby --version
+```
+If it is not found, run the follwing to install Ruby
+
+```{.sh}
+sudo apt-get install -y ruby
+```
+
+## Create an ERB template
+
+To make a distinction between ERB templates and normal sdformat files, `.sdf.erb` is used as file suffix. ERB language is usually embedded in the sdformat file. Below is a simple example to generate 1000 shapes using ERB templating language. Ruby syntax is used in the template part.
+
+```
+    <%
+      # number of population
+      n = 1000
+      for m in (0..n)
+    %>
+
+    <model name="box_<%= m.to_s %>">
```

Comment:
Consider adding an example that the user can run and adding detailed instructions on how to do it.

This snippet is helpful for users who already have intermediate SDF knowledge, but a beginner may have trouble plugging this into a workable model (it's missing the `<xml>` and `<sdf>` tags).

For example, you could create a working world and tell the user how to load it.